src/fsharp/FSharp.Core/array.fs

```diff
@@ -494,12 +494,23 @@ namespace Microsoft.FSharp.Collections
 
         [<CompiledName("Filter")>]
         let filter f (array: _[]) = 
-            checkNonNull "array" array
-            let res = List<_>() // ResizeArray
```

Comment:
Wouldn't we get a similar allocations benefit by merely pre-allocating this to array.length?  using List<_>(array.length)

https://msdn.microsoft.com/en-us/library/dw8e0z9z(v=vs.110).aspx